ironfish/src/account/accounts.ts

```diff
@@ -1118,12 +1118,20 @@ export class Accounts {
         const { blockHash } = transactionMapValue
 
         let status = 'pending'
+
         if (blockHash) {
           const header = await this.chain.getHeader(Buffer.from(blockHash, 'hex'))
           Assert.isNotNull(header)
-          const main = await this.chain.isHeadChain(header)
-          if (main) {
-            status = 'completed'
+
+          if (await this.chain.isHeadChain(header)) {
+            if (
+              this.chain.head.sequence - header.sequence <
+              this.config.get('minimumBlockConfirmations')
+            ) {
+              status = 'completed and finanlized'
```

Comment:
this should be backwards compatible - the completed state should still be called "completed", just in case someone is depending on that label

You can call the new state what you'd like, I would suggest "confirming block"

Reply:
Ok. Thanks @lwisne